spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/dynamodb/DaxProperties.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.dynamodb;
+
+/**
+ * Properties used to configure {@link software.amazon.dax.ClusterDaxClient}
+ * @author Matej Nedić
+ * @since 3.0.0
+ */
+public class DaxProperties {
+	/**
+	 * Timeout for idle connections with the DAX cluster.
+	 */
+	private int idleTimeoutMillis = 30000;
```

Comment:
I would prefer to avoid copying defaults from AWS SDK because then its easy to get out of sync. We should rather make all these fields as nullable and then in autoconfiguration use similar approach as we did with S3:

```java
PropertyMapper propertyMapper = PropertyMapper.get();
software.amazon.dax.Configuration.Builder configuration = software.amazon.dax.Configuration.builder();
propertyMapper.from(daxProperties.getIdleTimeoutMillis()).whenNonNull().to(configuration::idleTimeoutMillis);
```

Reply:
The problem with this is that `ClusterDaxClient.builder().overrideConfiguration()` copies each field. This means that we simply won't set the field then.
  ```
    this.configuration = (Configuration)configuration.copy((c) -> {
        });
```
Meaning users will have to set all of the fields inside of properties.